docs/Versions.md

```diff
@@ -0,0 +1,15 @@
+# Versions
+Now that [Homebrew/versions](https://github.com/homebrew/homebrew-versions) has been deprecated [Homebrew/core](https://github.com/homebrew/homebrew-core) now supports multiple versions of formulae with a new naming format.
+
+In [Homebrew/versions](https://github.com/homebrew/homebrew-versions) the formula for GCC 6 was named `gcc6.rb` and began `class Gcc6 < Formula`. In [Homebrew/core](https://github.com/homebrew/homebrew-core) this same formula is named `gcc@6.rb` and begins `class GccAT6 < Formula`.
+
+## Acceptable Versioned Formulae
+Homebrew's versions are not intended to be used for any old versions you personally require for your project; formulae submitted should be expected to be used by a large number of people and still supported by their upstream projects.
+
+You should create your own [tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md) for formulae you or your organisation wishes to control the versioning of or those that do not meet the above standards.
+
+Versioned formulae we include must meet the following standards:
+
+* Versioned formulae should differ in major/minor (not patch) versions from the current stable release. This is because patch versions indicate bug or security updates and we do not wish to allow you to avoid applying security updates.
+* Formulae that depend on versioned formulae must not depend on the same formulae at two different versions twice in their recursive dependencies. For example, if you depend on `openssl@1.0` and `foo` and `foo` depends on `openssl` then you must instead use `openssl`.
```

Comment:
> Can the version/head status of openssl as compared to openssl@1.0 be clarified here?

I don't understand what this means.

> When will versions simply be updated, and when will foo go from simply foo to foo@1.0, foo@2.0 and foo?

I'm not sure what this means but `foo` will always be the latest version. I may later add support for the latest version to also be accessed in `foo@2.0` format (i.e. if `foo` is currently 2.0.0).

> 3-6

We'll stick on older versions if it causes dependency issues. This is an issue that all package managers need to deal with and it's something Homebrew's been historically sloppy on. In short: if you package software, you need to figure this out. The current alternative is that you have software that will likely have hard-to-debug runtime errors.